src/transform/adm3.py

```diff
@@ -0,0 +1,62 @@
+import sys
+
+import geopandas as gpd
+import fiona
+from jsonschema import validate
+
+from utils.yaml_api import parse_yaml
+
+GADM_FILENAME = 'gadm36_{ISO3}.gpkg'
+GADM_LAYER = 'gadm36_{ISO3}_3'
+
+
+def transform_cod():
+    transform('cod', sys.argv[1], sys.argv[2], sys.argv[3])
+
+
+def transform_gadm():
+    transform('gadm', sys.argv[1], sys.argv[2], sys.argv[3])
+
+
+def transform(source: str, input_filename: str, schema_filename: str, output_filename: str):
+    """
+    :param source: "cod" or "gadm"
+    """
+    config = parse_yaml('config.yml')
+
+    if source == "cod":
+        layerlist = fiona.listlayers(f'zip://{input_filename}')
+        search = 'adm3'
+        for sublist in layerlist:
+            if search in sublist:
+                with fiona.open(f'zip://{input_filename}', layer=sublist) as layer:
+                    for feature in layer:
+                        if feature['geometry']['type'] == 'MultiPolygon':
+                            # print(feature['geometry']['type'],sublist)
+                            adm3 = sublist
```

Comment:
This method assumes that there is a single layer that satisifes the constraints. You could consider the case if there are multiple such layers, or none.